tools/lofi/probabilistic_get_ensembles_gcs.py

```diff
@@ -0,0 +1,166 @@
+import gc
+import os
+from datetime import datetime
+from typing import List
+
+import gcsfs
+import geopandas as gpd
+import numpy as np
+import xarray as xr
+from tqdm.notebook import tqdm
+
+
+nbm_ensemble_url = "gs://national-water-model/nwm.{0}/{1}/nwm.t{2}z.{1}.channel_rt.f{3}.conus.nc"
+gfs_ensemble_url = "gs://national-water-model/nwm.{0}/{1}/nwm.t{2}z.medium_range.channel_rt_{4}.f{3}.conus.nc"
+
+
+def get_gcs_ensembles(
+    dt: str, hour: str, ens_type: str, feature_ids: List[int], output_path: str, output_name: str
+):
+    """
+    Method to collect ensembles for NOMADS service
+
+    Parameters
+    ----------
+    dt : str
+        Date time string to get gcs forecast data, can be any date the product was available and account for the forecast window.
+    hour : str
+        Hours to get gcs forecast data.
+    ens_type : str
+        An ensemble type from the following list, "gfs", "nbm", "short", "noda".
+    feature_ids : List[int]
+        Feature IDs to get from the ensemble files.
+    output_path : str
+        Path to save ensemble files.
+    output_name : str
+        Name of final processed ensemble file.
+
+    """
+
+    gcs = gcsfs.GCSFileSystem()
+
+    if ens_type == "nbm":
+
+        forecast_type = 'medium_range_blend'
+
+        dt = datetime.strptime(f'{dt}-{hour}', '%Y%m%d-%H')
+        datetimes, hours = [dt.strftime('%Y%m%d')], [hour]
+        members = ['nbm']
+
+        # Get forecast times (3 hour timestep)
+        forecast_times = []
+        for num in np.arange(1, 241, 1):
+            num = str(num)
+            if len(num) == 2:
+                num = '0' + num
+            elif len(num) == 1:
+                num = '00' + num
+            forecast_times.append(num)
+
+        # Get nbm forecasts
+        ds_list = []
+        nofiles = []
+
+        for dtime, hr, mem in tqdm(zip(datetimes, hours, members)):
+
+            if len(str(hr)) < 2:
+                hr = f'0{hr}'
+
+            for ft in tqdm(forecast_times):
+
+                try:
+                    openfile = gcs.open(nbm_ensemble_url.format(dtime, forecast_type, hr, ft), mode='rb')
+                    dsgcs = xr.open_dataset(openfile)
```

Comment:
I believe you can pass the gcs URL directly to xarray. It will use fsspec to open the URL. It isn't necessary to open the file with gcsfs